### PR TITLE
doc: nrf5340audio: fixes from testing

### PR DESCRIPTION
**Pull request by greg-fer**
Added doc fixes from testing the new app split.

### CODE REVIEW

**Review by alexsven — APPROVED** [review]

**Comment by greg-fer** [comment]
@koffes , @Krrrl , can you review and approve? Thanks!

**Comment by koffes** [comment]
We already have two approvers from us, so I will not look at this unless it is required.

**Review by koffes — COMMENTED** [review]
We already have two approvers from us, so I will not look at this unless it is required.

File: doc/nrf/releases_and_maturity/software_maturity.rst [hunk 1]
```diff
@@ -114,7 +114,7 @@ The following table indicates the software maturity levels of the support for th
         - Description
         - Limitations
         - Maturity level
-      * - :ref:`**Broadcast source** <nrf53_audio_broadcast_source_app>`
+      * - :ref:`Broadcast source <nrf53_audio_broadcast_source_app>`
         - Broadcasting audio using Broadcast Isochronous Stream (BIS) and Broadcast Isochronous Group (BIG).
```
[the remark this answers is not in the document]
Comment (greg-fer) [reply doc/nrf/releases_and_maturity/software_maturity.rst:118]: I'll keep this as it is for now.

**Pull request merged**